System/DataModule.h

```diff
@@ -1,577 +1,348 @@
+		bool m_IgnoreMissingItems; //! Indicates whether module loader should ignore missing items in this module.
+
+		std::string m_FileName; //! File/folder name of the data module, eg "MyMod.rte".   
+		std::string m_FriendlyName; //! Friendly name of the data module, eg "My Weapons Mod". 
+		std::string m_Author; //! Name of the author of this module.
+		std::string m_Description; //! Brief description of what this module is and contains.
+		std::string m_ScriptPath; //! Path to script to execute when this module is loaded.
+		int m_Version; //! Version number, starting with 1.
+		int m_ModuleID; //! ID number assigned to this upon loading, for internal use only, don't reflect in ini's.
+
+		ContentFile m_IconFile; //! File to the icon/symbol bitmap.
+		BITMAP *m_pIcon; //! Bitmap with the icon loaded form above file.
+
+		float m_CrabToHumanSpawnRatio; //! Crab-to-human Spawn ratio to replace value from Constants.lua.
+
+		std::list<const Entity *> m_EntityList; //! A list of loaded entities solely for the purpose of enumeration presets from Lua.
+		std::list<std::string> m_GroupRegister; //! List of all Entity groups ever registered in this, all uniques.
+		unsigned char m_MaterialMappings[c_PaletteEntriesNumber]; //! Material mappings local to this DataModule.
+
+		/// <summary>
+		/// Ordered list of all owned Entity instances, ordered by the sequence of their reading - really now with overwriting?.
+		/// This is used to be able to write back all of them in proper order into their respective files in the DataModule when writing this.
+		/// The Entity instances ARE owned by this list.
+		/// </summary>
+		std::list<PresetEntry> m_PresetList;
+
+		/// <summary>
+		/// Map of class names and map of instance template names and actual Entity instances that were read for this DataModule.
+		/// An Entity instance of a derived type will be placed in EACH of EVERY of its parent class' maps here.
+		/// There can be multiple entries of the same instance name in any of the type sub-maps, but only ONE whose exact class is that of the type-list!
+		/// The Entity instances are NOT owned by this map.
+		/// </summary>
+		std::map<std::string, std::list<std::pair<std::string, Entity *> > > m_TypeMap;
+
+#pragma region Entity Mapping
+		/// <summary>
+		/// Checks if the type map has an instance added of a specific name and exact type.
+		/// Does not check if any parent types with that name has been added. If found, that instance is returned, otherwise 0.
+		/// </summary>
+		/// <param name="exactType">The exact type name to look for.</param>
+		/// <param name="instanceName">The exact instance name to look for.</param>
+		/// <returns>The found Entity Preset of the exact type and name, if found.</returns>
+		Entity * GetEntityIfExactType(const std::string &exactType, const std::string &instanceName);
+		//Entity * GetEntityIfExactType(const std::string &exactType, size_t instanceHash);
```

Comment:
Get rid of this commented line.

Also, I can't remember now if methods are supposed to be above members when they're in the same access scope. If so, please move these.
Actually now I think about it nothing inherits from datamodule so you could just make them private and basically nothing would change.